### ISSUE TITLE
As a graphics engine programmer I want to parallelise render sub-passes because I want to increase render performance

### ISSUE THREAD

**Issue by Zeeblur** [Zoeoeh/RayTracingTests#3]
- [ ] Is an abstract class for sub-passes defined?
- [ ] Do we have multiple sub-passes?
- [ ] Do we have deferred rendering?
- [ ] Do we have a shadow pass in deferred rendering?
- [ ] Can we run sub-passes in parallel?
- [ ] Are the sub-passes faster in parallel?
- [ ] Is the output from the parallel sub-passes the same as the sequential sub-passes?
